core/src/main/java/org/stellar/anchor/util/AssetValidator.java

```diff
@@ -0,0 +1,239 @@
+package org.stellar.anchor.util;
+
+import static java.lang.String.*;
+import static org.stellar.anchor.api.asset.Sep31Info.*;
+import static org.stellar.anchor.api.asset.Sep38Info.*;
+import static org.stellar.anchor.util.ListHelper.isEmpty;
+
+import java.util.*;
+import lombok.RequiredArgsConstructor;
+import org.stellar.anchor.api.asset.*;
+import org.stellar.anchor.api.exception.InvalidConfigException;
+import org.stellar.anchor.asset.AssetService;
+
+@RequiredArgsConstructor
+public class AssetValidator {
+  private static final List<String> isoCountries = Arrays.asList(Locale.getISOCountries());
+
+  public static void validate(AssetService assetService) throws InvalidConfigException {
+    // Check for non-zero assets
+    if (assetService == null || isEmpty(assetService.getAssets())) {
+      throw new InvalidConfigException("0 assets defined in configuration");
+    }
+
+    // Check for duplicate assets
+    Set<String> existingAssetNames = new HashSet<>();
+    for (AssetInfo asset : assetService.getAssets()) {
+      if (asset != null && !existingAssetNames.add(asset.getId())) {
+        throw new InvalidConfigException(
+            format("Duplicate assets defined in configuration. Asset = %s", asset.getId()));
+      }
+    }
+
+    // Validate stellar assets
+    for (StellarAssetInfo stellarAsset : assetService.getStellarAssets()) {
+      validateStellarAsset(assetService, stellarAsset);
+    }
+
+    // Validate fiat assets
+    for (FiatAssetInfo fiatAsset : assetService.getFiatAssets()) {
+      validateFiatAsset(assetService, fiatAsset);
+    }
+  }
+
+  // Check if deposit is enabled for the asset
+  public static boolean isDepositEnabled(DepositWithdrawInfo info) {
```

Comment:
We might want to put it under AssetHelper or AssetUtil?

Reply:
Moved.